client/src/browse/filterBar/FilterBar.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+
+const FilterBar = (props) => {
+  const [tags, setTags] = useState([]);
+  const [categories, setCategories] = useState([]);
+
+  useEffect(() => {
+    const newCategories = props.itemData.reduce(
+      (unique, item) =>
+        unique.includes(item.category) ? unique : [...unique, item.category],
+      []
+    );
+    setCategories(newCategories);
+
+    const newTags = [];
+    props.itemData.forEach((item) => {
+      item.tag.forEach((tag) => {
+        if (!newTags.includes(tag)) {
+          newTags.push(tag);
+        }
+      });
+    });
+    setTags(newTags);
+  }, [props.itemData]);
+
+  const filterItems = () => {
+    let filteredItems = props.itemData;
+    if (document.getElementById("distance").value !== "--Select Distance--") {
+      filteredItems = filteredItems.filter((item) => {
```

Comment:
Is there a reason you're looking up elements on the DOM instead of tracking with state?

Reply:
was just making it run, can change later. call it tech-debt for now?